File: app/src/main/java/org/witness/proofmode/service/OnBootReceiver.java
package org.witness.proofmode.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Created by n8fr8 on 10/10/16.
 */
public class OnBootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {

        if (android.os.Build.VERSION.SDK_INT >= 24) {
            //do nothing; the contentjobs are started in the app onCreate() right?
        }
        else
        {
            context.startService(new Intent(context, MediaListenerService.class));
        }
    }
}

File: app/src/main/java/org/witness/proofmode/util/OnBootReceiver.kt
package org.witness.proofmode.org.witness.proofmode.util

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.preference.PreferenceManager
import org.witness.proofmode.ProofMode

/**
 * Created by n8fr8 on 10/10/16.
 */
class OnBootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
        if (prefs.getBoolean(ProofMode.PREFS_DOPROOF, false)) {

            ProofMode.initBackgroundService(context)

            /**
            val intentService = Intent(context, ProofService::class.java)
            if (Build.VERSION.SDK_INT >= 26) {
                context.startForegroundService(intentService)
            } else {
                context.startService(intentService)
            }**/
        }
    }
}